File: SantiagoPinto_GenerarTiempos.cpp

#include<iostream>
#include <ctime>
using namespace std;

//devuelve el enesimo valor de la serie de fibonacci
int fibonacci(int N) {
    if(N < 2){
        return N;}
    else {
	return fibonacci(N-1) + fibonacci(N-2);}
  	}

//corre la serie de fibonacci
float get_time(int N) {	 
  clock_t t;
  t = clock();

// SOME CODE THAT TAKES TIME
  fibonacci(N);

  t = clock() - t;
  float time = ((float)t)/CLOCKS_PER_SEC;
  return time;}



//corre todo
int main() {
  int N;
   N=35;
  fibonacci(N); 
  get_time(N);
  
  for(int i =0; i <= N; i++)
  {cout << i <<" "<< get_time(i) << " "<< endl;}


  return 0;
}

File: SantiagoPinto_GenerarTiempos.py
import time
# n termino
N=35

#ESTA FUNCION DEVUELVE EL ENECIMO TERMINO DE la serie de fibonacci
def fibonacci(N):
	if(N < 2):
        	return N
	else:
		return fibonacci(N-1) + fibonacci(N-2)

#esta funcion devuelve el tiempo que se demora en calcular la maquina el enecimo termino de la serie de fibonacci
  	
def get_time(N):
	
	t0 = time.time()
	fibonacci(N)
	t1 = time.time()-t0
	return t1

for i in range (N+1):
	print i, get_time(i)
